pkg/api/controller_test.go

```diff
@@ -480,6 +480,24 @@ func TestController_LogCommitsPredefinedData(t *testing.T) {
 			expectedMore:    false,
 			stopAt:          commits[8].Reference,
 		},
+		{
+			name:            "stop_at_short_sha",
+			expectedCommits: []string{"commit10", "commit9"},
+			expectedMore:    false,
+			stopAt:          commits[8].Reference[:7],
+		},
+		{
+			name:            "stop_at_branch_ref",
+			expectedCommits: []string{"commit10"},
+			expectedMore:    false,
+			stopAt:          "main",
+		},
+		{
+			name:            "stop_at_branch_ref_expression",
+			expectedCommits: []string{"commit10", "commit9", "commit8"},
+			expectedMore:    false,
+			stopAt:          "main~2",
+		},
```

Comment:
Honestly it's cool, but I see no reason to test each of these.  Any one of these tests proves that we Deref the ref.  A unit test of logging does not need to do anything beyond testing that this happens.  (All of these -- and more -- are tested in `parse_ref_test.go` and `resolve_ref_test.go`!)

Reply:
But it doesn't test whether we dereference the StopAt flag.
If we were testing it, I wouldn't be able to merge my [PR](https://github.com/treeverse/lakeFS/pull/7222)